include/caffe/layer.hpp

```diff
@@ -413,7 +413,7 @@ template <typename Dtype>
 inline Dtype Layer<Dtype>::Forward(const vector<Blob<Dtype>*>& bottom,
     const vector<Blob<Dtype>*>& top) {
   Dtype loss = 0;
-  Reshape(bottom, top);
+  // Reshape(bottom, top);
```

Comment:
I think the Reshape in Layer is necessary for those cases with inputs with different size in a batch. Also, the Reshape will take almost zeros cost if the blob is in correct shape. Hence, I do not think it is necessary to remove this line.